bdsx/launcher.ts

```diff
@@ -401,15 +422,30 @@ export namespace bedrockServer {
      * it does the same thing with executeCommandOnConsole
      * but call the internal function directly
      */
-    export function executeCommand(command:string, mute:boolean=true, permissionLevel:number=4, dimension:Dimension|null = null):MCRESULT {
+    export function executeCommand(command:string, mute:boolean=true, permissionLevel:number=4, dimension:Dimension|null = null, callback?:(data:IExecuteCommandCallback["data"]) => void):MCRESULT {
         const minecraft = bd_server.serverInstance.minecraft;
         const origin = ServerCommandOrigin.allocateWith('Server',
             minecraft.getLevel() as ServerLevel, // I'm not sure it's always ServerLevel
             permissionLevel,
             dimension);
         const ctx = CommandContext.constructSharedPtr(command, origin);
-        const res = minecraft.getCommands().executeCommand(ctx, mute);
+        const uuid = origin.uuid;
+        commandOutputData.set(uuid, null);
+        if (mute) commandOutputMuted.add(uuid); // Mute it after statusMessage is gotten
+        const res = minecraft.getCommands().executeCommand(ctx, false); // Do not mute it, otherwise there will be no statusMessage
         // ctx, origin: no need to destruct, it's destructed by internal functions.
+        if (callback) {
+            const json = commandOutputData.get(uuid)!;
+            commandOutputData.delete(uuid);
```

Comment:
if `callback` is not provided. `commandOutputData` will increase without deleting.

Reply:
Oh I see, I missed that